### PR TITLE
dtd_bench_simple_gemm.c relies on non-standard cblas.h file

### PR DESCRIPTION
**Pull request by therault**
to compute GEMMs on CPUs, if a BLAS library is discovered
by CMake.

Define the prototype of the dgemm call inside the file, as BLAS
does not define a include file and depending on the BLAS library
found this file might exist or not.

Also remove a deprecated function call warning that was not captured
in previous commits.

### CODE REVIEW

File: tests/dsl/dtd/dtd_bench_simple_gemm.c [hunk 1]
```diff
@@ -4,18 +4,32 @@
 #include "parsec/data_dist/matrix/two_dim_rectangle_cyclic.h"
 #include "parsec/interfaces/dtd/insert_function_internal.h"
 #include "cublas_v2.h"
-#if HAVE_BLAS
-#if BLAS_WITH_ESSL
-#include "essl.h"
+
+#if _HAVE_ACCELERATE_H
+#include<Accelerate/Accelerate.h>
+#elif _HAVE_COREBLAS_H
```
Comment (abouteiller) [inline tests/dsl/dtd/dtd_bench_simple_gemm.c:10]: These defines are not set anywhere. What's going on?

File: tests/dsl/dtd/dtd_bench_simple_gemm.c [hunk 2]
```diff
@@ -4,18 +4,32 @@
...
+#include <cblas.h>
+#elif _HAVE_MKLBLAS_H
+#include <mkl_cblas.h>
+#elif _HAVE_ESSL_H
+#include <essl.h>
 #else
-#include "cblas.h"
-#endif
+typedef enum CBLAS_LAYOUT {CblasRowMajor=101, CblasColMajor=102} CBLAS_LAYOUT;
```
Comment (bosilca) [inline tests/dsl/dtd/dtd_bench_simple_gemm.c:17]: Only keep those and do not include any cblas header.

File: tests/dsl/dtd/dtd_bench_simple_gemm.c [hunk 3]
```diff
@@ -4,18 +4,32 @@
 #include "parsec/data_dist/matrix/two_dim_rectangle_cyclic.h"
```
Comment (bosilca) [inline tests/dsl/dtd/dtd_bench_simple_gemm.c:4]: Rename to not have bench in the file name.

**Comment by therault** [comment]
This is ready for review

**Review by bosilca — APPROVED** [review]

**Review by abouteiller — APPROVED** [review]

**Pull request merged**